### PR TITLE
[MOS-402] Time legibility on lock screen fix

### PR DESCRIPTION
**Pull request by Lefucjusz**
**Description**

Fix of the issue that time on the lock screen was
getting illegible after a few hours. This change
fixes the problem by performing deep refresh
instead of fast refresh every 30 minutes. The
refresh rate can be easily changed, as it is stored
in settings database.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [x] Has unit tests if possible.
- [x] Has documentation updated

### CODE REVIEW

File: module-apps/apps-common/ApplicationCommon.cpp [hunk 1]
```diff
@@ -1058,4 +1060,37 @@ namespace app
...
+        }
+        else {
+            LOG_DEBUG("Requesting fast refresh on phone lock screen");
+            refreshWindow(gui::RefreshModes::GUI_REFRESH_FAST);
+        }
+    }
+
+    void ApplicationCommon::initializeDeepRefreshCounter()
+    {
+        constexpr auto fallbackRefreshRate = 15;
```
Comment (SP2FET) [inline module-apps/apps-common/ApplicationCommon.cpp:1080]: why fallback is different than the DB value?
consider using `std::chrono`'s literals

**Review by SP2FET — APPROVED** [review]

File: Jenkinsfile [hunk 2]
```diff
@@ -13,7 +13,7 @@ pipeline {
     }
     environment {
         JOBS=30
-        PATH="/usr/local/cmake-3.21.3-linux-x86_64/bin:/usr/local/gcc-arm-none-eabi-10-2020-q4-major/bin:$PATH"
+        PATH="/usr/local/cmake-3.21.3-linux-x86_64/bin:/usr/local/gcc-arm-11.2-2022.02-x86_64-arm-none-eabi/bin:$PATH"
```
Comment (pholat) [inline Jenkinsfile:16]: this shouldn't be here?

File: module-apps/apps-common/ApplicationCommon.cpp [hunk 3]
```diff
@@ -468,12 +468,13 @@ namespace app
     sys::MessagePointer ApplicationCommon::handleMinuteUpdated(sys::Message *msgl)
     {
         if (state == State::ACTIVE_FORGROUND && getCurrentWindow()->updateTime()) {
-
             if (isOnPhoneLockWindow()) {
                 updateStatusBarOnPhoneLockWindow();
+                refreshTimeOnPhoneLockWindow();
```
Comment (pholat) [inline module-apps/apps-common/ApplicationCommon.cpp:473]: why not:
- override handleMinuteUpdated in the PhoneLockWindow
- change return type there
It would remove Window specific code fully from the ApplicationComomn

File: module-apps/apps-common/ApplicationCommon.cpp [hunk 4]
```diff
@@ -1058,4 +1060,37 @@ namespace app
...
+        if (lockScreenRefreshCounter.currentValue >= lockScreenRefreshCounter.referenceValue) {
+            lockScreenRefreshCounter.currentValue = 0;
+            LOG_DEBUG("Requesting deep refresh on phone lock screen");
+            refreshWindow(gui::RefreshModes::GUI_REFRESH_DEEP);
+        }
+        else {
+            LOG_DEBUG("Requesting fast refresh on phone lock screen");
+            refreshWindow(gui::RefreshModes::GUI_REFRESH_FAST);
+        }
+    }
```
Comment (pholat) [inline module-apps/apps-common/ApplicationCommon.cpp:1076]: move check to the struct  and handle refresh separately? similar below

File: module-apps/apps-common/popups/lock-popups/PhoneLockedWindow.hpp [hunk 5]
```diff
@@ -18,24 +18,52 @@ namespace gui
...
 
         std::unique_ptr<WallpaperPresenter> wallpaperPresenter;
         bool refreshedOnPhoneLockTimeLock = false;
+
+        void initializeDeepRefreshCounter(std::uint8_t deepRefreshRate);
+
+        struct
+        {
+            std::uint8_t currentValue;
+            std::uint8_t referenceValue;
```
Comment (pholat) [inline module-apps/apps-common/popups/lock-popups/PhoneLockedWindow.hpp:43]: why not plain `unsigned int` ? It's not like we care about integer precision here ;) 

small nitpick - do whatever you want with this comment :bowing_man:

File: module-apps/apps-common/ApplicationCommon.cpp [hunk 6]
```diff
@@ -467,13 +467,13 @@ namespace app
 
     sys::MessagePointer ApplicationCommon::handleMinuteUpdated(sys::Message *msgl)
     {
-        if (state == State::ACTIVE_FORGROUND && getCurrentWindow()->updateTime()) {
-
+        if (state == State::ACTIVE_FORGROUND) {
+            auto requestedRefreshMode = getCurrentWindow()->updateTime();
             if (isOnPhoneLockWindow()) {
                 updateStatusBarOnPhoneLockWindow();
             }
```
Comment (pholat) [inline module-apps/apps-common/ApplicationCommon.cpp:474]: didn't we agree that it can be moved to the `PhoneLockWindow` and both these methods can be removed from the ApplicationCommon? :) It's not much, but when we can remove unrelated code from "common" parts - it's worth it ;)

**Review by pholat — APPROVED** [review]
Please remove the code that is not required and can be solved with existing inheritance, then close the comments and merge. Good job :beer:

File: module-apps/apps-common/ApplicationCommon.cpp (hunk 1 above)
Comment (Lefucjusz) [reply module-apps/apps-common/ApplicationCommon.cpp]: My mistake, fixed now.
As for literals, I don't think using them here is required. I'm only counting calls of this function to count the time, not using some "real" timer ;)

File: Jenkinsfile (hunk 2 above)
Comment (Lefucjusz) [reply Jenkinsfile]: True, it was added to test some CI-related issue, removed it now

File: module-apps/apps-common/popups/lock-popups/PhoneLockedWindow.hpp (hunk 5 above)
Comment (Lefucjusz) [reply module-apps/apps-common/popups/lock-popups/PhoneLockedWindow.hpp]: Just a habit from coding for smaller, 8-bit CPUs :P Changed to unsigned.

File: module-apps/apps-common/ApplicationCommon.cpp (hunk 6 above)
Comment (Lefucjusz) [reply module-apps/apps-common/ApplicationCommon.cpp]: isOnPhoneLockWindow() is used in multiple different places, so I left this method, but updateStatus... content was moved to PhoneLockedWindow as you suggested ;)

**Review by pholat — APPROVED** [review]

File: module-apps/apps-common/ApplicationCommon.cpp (hunk 6 above)
Comment (pholat) [reply module-apps/apps-common/ApplicationCommon.cpp]: tx, I'll see if it could be possibly removed in other places too :)
